From 2c60225fd559603170cd32f84be06572c1b96c67 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 20 Nov 2018 13:36:31 +0800
Subject: [PATCH 5153/5242] MLK-20140 ARM64: dts: freescale: imx8mm-ddr4-evk:
 correct LED GPIO pin

commit  4a8f5b890899e6ac29eba3d82369436f27450eaa from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX8MM DDR4 EVK board, the GPIO pin to control LED is
NAND_CE3_B, correct it to make LED work properly and avoid
below failure message during kernel boot up:

[ 3.274994] leds-gpio leds: Error applying setting, reverse things back
[ 3.275015] leds-gpio: probe of leds failed with error -22

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts     |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts
index ee54c1c..6eb227b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-ddr4-evk.dts
@@ -9,6 +9,14 @@
 
 / {
 	model = "FSL i.MX8MM DDR4 EVK board";
+
+	leds {
+		pinctrl-0 = <&pinctrl_gpio_led_2>;
+
+		status {
+			gpios = <&gpio3 4 GPIO_ACTIVE_LOW>;
+		};
+	};
 };
 
 &iomuxc {
@@ -33,6 +41,12 @@
 				MX8MM_IOMUXC_NAND_WP_B_RAWNAND_WP_B		0x00000096
 			>;
 		};
+
+		pinctrl_gpio_led_2: gpioled2grp {
+			fsl,pins = <
+				MX8MM_IOMUXC_NAND_CE3_B_GPIO3_IO4	0x19
+			>;
+		};
 	};
 };
 
-- 
1.7.9.5

